speculation-rules.bs

```diff
@@ -217,6 +233,85 @@ Inside the [=prepare the script element=] algorithm we make the following change
     </dd>
   </dl>
 
+<h3 id="speculation-rules-header">The [:Speculation-Rules:] header</h3>
+Note: This section contains modifications to [[HTML]].
+
+<div algorithm="process the Speculation-Rules header">
+
+  To <dfn>process the Speculation-Rules header</dfn> given a [=document=] |document| and a [=response=] |response|:
+
+  1. Optionally, return.
+    <p class="note">The user agent could ignore this header, for example, if it does not intend to [=consider speculation=] for |document|.</p>
+  1. Let |parsedList| be the result of [=header list/getting a structured field value=] given [:Speculation-Rules:] and "`list`" from |response|'s [=response/header list=].
+  1. If |parsedList| is null, then return.
+  1. Let |baseUrl| be |document|'s [=document base URL=].
+  1. [=list/For each=] |item| of |parsedList|:
+    1. If |item| is not a [=string=], then [=iteration/continue=].
+    1. Let |parsedURL| be the result of [=basic URL parser|parsing=] |item| with |baseUrl|.
+    1. If |parsedURL| is failure, then [=iteration/continue=].
+    1. Let |pendingResource| be a [=pending external speculation rule resource=] with [=pending external speculation rule resource/URL=] |parsedURL| and [=pending external speculation rule resource/controller=] null
+    1. [=list/Append=] |pendingResource| to |document|'s [=document/list of pending external speculation rule resources=].
+  1. [=Process pending external speculation rule resources=] given |document|.
+</div>
+
+In [=shared document creation infrastructure=], before the step
+
+  > 19. Return <var ignore>document</var>.
+
+add the following step
+
+  > 19. [=process the Speculation-Rules header|Process the Speculation-Rules header=] given <var ignore>document</var> and <var ignore>navigationParams</var>'s [=response=].
+
+<h3 id="external-speculation-rule-sets">External speculation rule sets</h3>
+
+<div algorithm="process pending external speculation rule resources">
+
+  To <dfn>process pending external speculation rule resources</dfn> given a [=document=] |document|:
+
+  1. Let |pendingResources| be |document|'s [=document/list of pending external speculation rule resources=].
+  1. [=list/For each=] |pendingResource| of |pendingResources|:
+    1. Optionally, [=pending external speculation rule resource/cancel and discard=] |pendingResource| given |document|, then [=iteration/continue=].
+    1. If |pendingResource|'s [=pending external speculation rule resource/controller=] is not null, then [=iteration/continue=].
+    1. Optionally, [=pending external speculation rule resource/fetch=] |pendingResource| given |document|.
+      <p class="note">The user agent may schedule the fetch of external speculation rules at its discretion.</p>
```

Comment:
Maybe to make it extra clear, say something like. "That is, if it skips the fetch during this call of [process pending external speculation rule resources], it might do the fetch during a later call."

Reply:
Done.